### PR TITLE
modules: hostap: add tls_cipher param

### PR DESCRIPTION
**Pull request by LiLongNXP**
Add tls_cipher param for client WPA3 enterprise suiteb-192. Add parameter "-T" to specify tls_cipher:
Specify "-T 1": client use ECC P384.
Specify "-T 2": client use RSA 3K.

### CODE REVIEW

**Comment by LiLongNXP** [comment]
hi @krish2718 @jukkar ,
Could you please review and merge this PR if it reviewed done.

This PR depend on below 2 PRs which also hasn't been merged:
https://github.com/zephyrproject-rtos/hostap/pull/20
https://github.com/zephyrproject-rtos/mbedtls/pull/62

Thanks.

**Comment by LiLongNXP** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/80214

> hi @krish2718 @jukkar , Could you please review and merge this PR if it reviewed done.
> 
> This PR depend on below 2 PRs which also hasn't been merged: [zephyrproject-rtos/hostap#20](https://github.com/zephyrproject-rtos/hostap/pull/20) [zephyrproject-rtos/mbedtls#62](https://github.com/zephyrproject-rtos/mbedtls/pull/62)
> 
> Thanks.

hi @krish2718 , @jukkar ,
Since the original WAR PR (pull/67) as blow is merged instead of above TODO PR (pull/20 pull/62).
https://github.com/zephyrproject-rtos/hostap/pull/67

Then this PR is not blocked now, please review and if ok merge this PR.

Thanks,
Li

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -785,6 +786,9 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 		case 'S':
 			params->suiteb_type = atoi(state->optarg);
 			break;
+		case 'T':
+			params->TLS_cipher = atoi(optarg);
```
Comment (MaochenWang1) [inline subsys/net/l2/wifi/wifi_shell.c:790]: Please use `state->optarg`, same as other lines in this function

Comment (LiLongNXP) [reply subsys/net/l2/wifi/wifi_shell.c]: Thanks. Code updated.

**Review by MaochenWang1 — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**